### PR TITLE
Release notes for 0.3.10

### PR DESCRIPTION
**Pull request by peterschwarz**

### CODE REVIEW

**Review by agunde406 — APPROVED** [review]

**Review by vaporos — DISMISSED** [review]

File: RELEASE_NOTES.md [hunk 1]
```diff
@@ -1,5 +1,33 @@
 # Release Notes
 
+## Highlights
+
+* The `"state-merkle-leaf-reader"` feature has been stabilize by being removed.
```
Comment (agunde406) [inline RELEASE_NOTES.md:7]: stabilize -> stabilized

File: RELEASE_NOTES.md [hunk 2]
```diff
@@ -1,5 +1,33 @@
...
+  public API.
+
+* Soft-deprecate the type `transact::state::merkle::kv::MerkelLeafIterator`, as
+  this type should not be part of the public API, and may be removed in a future
+  release.
+
+* Return `InvalidStateError` when a `StateDatabaseError::NotFound` variant is
+  encountered during leaf iteration when implemented on `kv::MerkleState`.
+
+* The `"state-merkle-leaf-reader"` feature has been stabilize by being removed.
```
Comment (agunde406) [inline RELEASE_NOTES.md:28]: Wrong format for this section, should say something like Stabilize `"state-merkle-leaf-reader" feature.

**Review by agunde406 — CHANGES_REQUESTED** [review]

File: RELEASE_NOTES.md (hunk 1 above)
Comment (peterschwarz) [reply RELEASE_NOTES.md]: Fixed

File: RELEASE_NOTES.md (hunk 2 above)
Comment (peterschwarz) [reply RELEASE_NOTES.md]: Fixed

**Review by vaporos — APPROVED** [review]

**Review by agunde406 — APPROVED** [review]

**Pull request merged**